fuzz/fuzz_targets/message_decode.rs

```diff
@@ -0,0 +1,10 @@
+#![no_main]
+
+use libfuzzer_sys::fuzz_target;
+use ssh_agent_lib::proto::message::Message;
+use ssh_encoding::Decode;
+
+fuzz_target!(|data: &[u8]| {
+    let mut data = data;
+    let _ = Message::decode(&mut data);
```

Comment:
it likely won't have any effect on the compiled code, but you could write this as:

```suggestion
    let _ = Message::decode(&mut &data[..]);
```

or by writing `|mut data: &[u8]| {`

Reply:
I've used your suggestion and squashed the commit (with your name in `Co-authored-by` trailer). the `|mut data..|` didn't seem to work due to the issue with the `fuzz_target!` macro... but the other ref-magic works fine :ok_hand: